### PR TITLE
DEV: Remove HTML setting type and sanitization logic.

### PR DESCRIPTION
**Pull request by romanrizzi**
We concluded that we don't want settings to contain HTML, so I'm removing the setting type and sanitization logic. Additionally, we no longer allow the global-notice text to contain HTML.

I searched for usages of this setting type in the `all-the-plugins` repo and found none, so I haven't added a migration for existing settings.

### CODE REVIEW

**Comment by CvX** [comment]
> I searched for usages of this setting type in the all-the-plugins repo and found none

Just to confirm: also nothing in all-the-themes, and in the customer-variant repos of both types?

File: lib/site_settings/type_supervisor.rb [hunk 1]
```diff
@@ -37,7 +37,6 @@ def self.types
       color: 22,
       simple_list: 23,
       emoji_list: 24,
-      html: 25
```
Comment (CvX) [inline lib/site_settings/type_supervisor.rb:40]: As it is with all enum types - I'm going to ask: do we want to re-use that number in the future or do we want to leave a comment/rename that to e.g. `html_deprecated`?

**Comment by romanrizzi** [comment]
> > I searched for usages of this setting type in the all-the-plugins repo and found none
> 
> Just to confirm: also nothing in all-the-themes, and in the customer-variant repos of both types?

Yes, checked all the four repos. I already refactored the only plugin that was using this type.

> As it is with all enum types - I'm going to ask: do we want to re-use that number in the future or do we want to leave a comment/rename that to e.g. html_deprecated?

I understand there's some level of risk, but is there a reason why the number cannot be re-used? The type only existed for a relatively short period, so no one got the chance to use it.

**Review by CvX — APPROVED** [review]
👍

File: lib/site_settings/type_supervisor.rb (hunk 1 above)
Comment (tgxworld) [inline lib/site_settings/type_supervisor.rb:40]: If this type is no longer valid, we need a migration to clean it up on the database.

Comment (romanrizzi) [reply lib/site_settings/type_supervisor.rb:40]: I added the migration and renamed the type to `html_deprecated` just to be safe. Could you please take another look?

File: db/migrate/20210928161912_migrate_deprecated_html_setting_type.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def up
+    execute <<~SQL
+      UPDATE site_settings
+      SET data_type = 1
+      WHERE data_type = 25
+    SQL
+  end
+
+  def down
+  end
```
Comment (tgxworld) [inline db/migrate/20210928161912_migrate_deprecated_html_setting_type.rb:13]: For consistency with other migrations, let's raise a irreversible migration error here.

**Review by tgxworld — APPROVED** [review]

**Comment by tgxworld** [comment]
@romanrizzi The migration looks good to me :+1:

**Pull request merged**